Source/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -90,10 +93,33 @@ public ISetupSequentialResult<TResult> CallBase()
 
 		public ISetupSequentialResult<TResult> Returns(TResult value)
 		{
+			// If `TResult` is `Func<>`, that is someone is setting up the return value of a method
+			// that returns a `Func<>`, then we need to make sure that the passed Func will be treated 
+			// as a return value. We don't want to invoke the passed Func to get a return value; the 
+			// passed func already is the return value. To prevent it we need to wrap up this func because 
+			// `SequenceMethodCall.Execute` invokes every func it encounters
+			if (typeof(TResult).IsConstructedGenericType && typeof(TResult).GetGenericTypeDefinition() == typeof(Func<>))
```

Comment:
In the latest changes I tried to cover slightly different case of breaking changes:

After my changes `SequenceMethodCall.Execute()` treats all `Delegate` as a function that need to be evaluated. Because of these an old code that mocked method like `Func<AnyType> GetFunc()` would be broken. So it's why I have a check inside `Returns(TResult)`. Although my checks didn't cover all cases.

Now I am considering that it's better to have a new enum value `SequenceMethodCall.ResponseKind.InvokeFunc` for all funcs that need to be invoked to get the return value. It will allow to remove checks that I was trying to accomplish.

Anyway I am still missing the scenario that you described. If I understand it right we also need to handle some other cases:
```
                public interface IFoo
 		{ 
 			object GetObj();

			Delegate GetDel();

                        MulticastDelegate GetMulticastDel()
                 }
```